clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -183,4 +197,50 @@ public synchronized long lastSuccessfulUpdate() {
     public long refreshBackoff() {
         return refreshBackoffMs;
     }
+
+    /**
+     * Set state to indicate if metadata for all topics in Kafka cluster is required or not.
+     * @param needMetadaForAllTopics boolean indicating need for metadata of all topics in cluster.
+     */
+    public void needMetadataForAllTopics(boolean needMetadaForAllTopics) {
+        this.needMetadataForAllTopics = needMetadaForAllTopics;
```

Comment:
Haha.. I actually made the changes and realized the same.. It does make it look cleaner though. Also, one more issue would be is when subscribed via pattern, metadata will always store metadata for all topics and a fetch will return metadata for all topics, not just for the topics that are actually subscribed via pattern subscription.

For now, I think we can just leave it as it is. If you think we should refactor partitionsFor() then let me know, we probably should create a separate JIRA for it though.